src/core/communication.cpp

```diff
@@ -1864,7 +1864,7 @@ void mpi_send_exclusion_slave(int part1, int part2) {
 }
 
 /************** REQ_SET_FLUID **************/
-void mpi_send_fluid(int node, int index, double rho, double *j, double *pi) {
+void mpi_send_fluid(int node, int index, double rho, const std::array<double, 3> j, const std::array<double, 6> pi) {
```

Comment:
Is there a specific reason to take copies of j and pi here?

Reply:
no, fixed